src/Orleans.Transactions/State/TransactionalState.cs

```diff
@@ -120,11 +121,8 @@ IClock clock
             lockWorker = new BatchWorkerFromDelegate(LockWork);
             storageWorker = new BatchWorkerFromDelegate(StorageWork);
             confirmationWorker = new BatchWorkerFromDelegate(ConfirmationWork);
-
-            if (MetaData.SerializerSettings == null)
-            {
-                MetaData.SerializerSettings = TransactionParticipantExtensionExtensions.GetJsonSerializerSettings(typeResolver, grainFactory);
-            }
+            
+            this.serializerSettings = TransactionParticipantExtensionExtensions.GetJsonSerializerSettings(typeResolver, grainFactory);
```

Comment:
Suggest serializerSettings be passed in through constructor.  Push responsibility up a layer to the factory.

Advantages:
- Reduces dependencies, as typeResolver and grainFactory are only passed in to construct these settings.
- Improves testability, due to reduced dependencies, and responsivity.
- More flexible, as users can replace factory (not complicated) to customize xml settings.

Reply:
will do